core/src/main/java/nl/nn/adapterframework/pipes/PGPPipe.java

```diff
@@ -0,0 +1,181 @@
+ * you can seperate multiple values with ";" (semicolon).
+ * </p>
+ */
+public class PGPPipe extends FixedForwardPipe {
+	/**
+	 * Action to be taken by pipe.
+	 * Available Actions:
+	 * <ul>
+	 *     <li>Encrypt: Encrypts the given input</li>
+	 *     <li>Sign: Encrypts and then signs the given input</li>
+	 *     <li>Decrypt: Decrypts the given input</li>
+	 *     <li>Verify: Decrypts and verifies the given input</li>
+	 * </ul>
+	 */
+	private String action;
+	/**
+	 * Emails of the recipients
+	 */
+	private String[] recipients;
+	/**
+	 * Emails of the senders. This will be used to verify that all the senders have signed the given message.
+	 * If not set, and the action is verify; this pipe will validate that at least one person has signed.
+	 *
+	 * For signing action, it needs to be set to the email that was used to generate the private key
+	 * that is being used for this process.
+	 */
+	private String[] senders;
+	/**
+	 * Path to the private key. It will be used when signing or decrypting.
+	 */
+	private String secretKey;
+	/**
+	 * Password for the private key.
+	 */
+	private String secretPassword;
+	/**
+	 * Path to the recipient's public key. It will be used for encryption.
+	 */
+	private String[] publicKey;
+
+	/**
+	 * This is the {@link PGPAction} object that executes the desired action.
+	 */
+	private PGPAction pgpAction;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+		if (action == null)
+			throw new ConfigurationException("Action can not be null!");
+
+		if(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) != null)
+			Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
+
+		Security.addProvider(new BouncyCastleProvider());
+
+		switch (action.toLowerCase()) {
+			case "encrypt":
+				pgpAction = new Encrypt(publicKey, recipients);
+				break;
+			case "decrypt":
+				pgpAction = new Decrypt(secretKey, secretPassword);
+				break;
+			case "sign":
+				if(senders == null || senders.length == 0)
+					throw new ConfigurationException("During signing action, senders has to be set.");
+				pgpAction = new Sign(publicKey, secretKey, secretPassword, recipients, senders[0]);
+				break;
+			case "verify":
+				pgpAction = new Verify(publicKey, secretKey, secretPassword, senders);
+				break;
+			default:
+				throw new ConfigurationException("Unknown action. Action has to be set to one of [Encrypt, Decrypt, Sign, Verify]");
+		}
+		pgpAction.configure();
+	}
+
+	@Override
+	public PipeRunResult doPipe(Object input, IPipeLineSession session) throws PipeRunException {
+		Message message = new Message(input);
+		try {
+			OutputStream outputStream = pgpAction.run(message.asInputStream());
+			return new PipeRunResult(findForward("success"), outputStream);
+		} catch (Exception e) {
+			throw new PipeRunException(this, "Exception was thrown during PGPPipe execution.", e);
+		}
+	}
+
+	public void setAction(String action) {
```

Comment:
Can you add the IbisDoc annotation for all setters, so the ibisdevelopers can see what all these things do?